### PR TITLE
Support MapView in foss flavor

### PR DESCRIPTION
**Pull request by mueller-ma**
See #574

Signed-off-by: mueller-ma <mueller-ma@users.noreply.github.com>

### CODE REVIEW

File: mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.java [hunk 1]
```diff
@@ -1,13 +1,255 @@
...
+        }
+
+        /**
+         * This method converts dp unit to equivalent pixels, depending on device density.
+         *
+         * @param dp A value in dp (density independent pixels) unit. Which we need to convert into pixels
+         * @param context Context to get resources and device specific display metrics
+         * @return A float value to represent px equivalent to dp depending on device density
+         */
+        private static float convertDpToPixel(float dp, Context context){
```
Comment (mueller-ma) [inline mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.java]: I'll use the copy introduced in https://github.com/openhab/openhab-android/pull/1357/files#diff-6a62feee90062d42673ee06ee0626d43R384.

File: mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.java [hunk 2]
```diff
@@ -1,13 +1,255 @@
...
+                    return true;
+                }
+
+                @Override
+                public boolean longPressHelper(GeoPoint p) {
+                    return false;
+                }
+            }));
+
+            new Handler().post(() -> applyPositionAndLabel(mMapView, 15.0f, false, false));
```
Comment (maniac103) [inline mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.java]: Is the handler really needed? If so, should probably make it a member variable considering it's used multiple times.

File: mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.java [hunk 3]
```diff
@@ -1,13 +1,255 @@
...
+
+            AlertDialog dialog = new AlertDialog.Builder(itemView.getContext())
+                    .setView(mapView)
+                    .setCancelable(true)
+                    .setNegativeButton(R.string.close, null)
+                    .create();
+
+            dialog.setOnDismissListener(dialogInterface -> {
+                mapView.onPause();
+            });
```
Comment (maniac103) [inline mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.java]: Can shorten this to one line.

File: mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.java (hunk 2 above)
Comment (mueller-ma) [reply mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.java]: If I don't use the Handler here, the markers are applied before the MapView is inflated, leading to runtime crashes.

Comment (maniac103) [reply mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.java]: I can not reproduce that using the emulator. How does the crash trace look like for you?

Comment (mueller-ma) [reply mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.java]:
```
05-10 22:53:23.356 org.openhab.habdroid.beta D/MapViewHelper: North 52.516998, south 52.506340, west 10.388490, east 15.388490
05-10 22:53:23.357 org.openhab.habdroid.beta D/AndroidRuntime: Shutting down VM
05-10 22:53:23.361 org.openhab.habdroid.beta E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.openhab.habdroid.beta, PID: 7685
    java.lang.IllegalArgumentException: north must be in [-85.05112877980658,85.05112877980658]
        at org.osmdroid.util.BoundingBox.set(BoundingBox.java:67)
        at org.osmdroid.views.Projection.refresh(Projection.java:680)
        at org.osmdroid.views.Projection.<init>(Projection.java:99)
        at org.osmdroid.views.Projection.<init>(Projection.java:110)
        at org.osmdroid.views.MapView.zoomToBoundingBox(MapView.java:556)
        at org.osmdroid.views.MapView.zoomToBoundingBox(MapView.java:582)
        at org.openhab.habdroid.ui.MapViewHelper$OsmViewHolder.applyPositionAndLabel(MapViewHelper.java:207)
```

Opened an issue here: https://github.com/osmdroid/osmdroid/issues/1339#issuecomment-491487934

This only occurs if you have more than one markers and remove all the Handlers and the try/catch below.

Comment (maniac103) [reply mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.java]: Ah, OK, only tried with a single item. Thanks for that link giving context; let's see what they'll say. The `Handler` seems like a workaround for an issue on their side ... and if we want to keep it, we might rather attach a `ViewTreeObserver.OnGlobalLayoutListener` to see whether the map is already laid out.

Comment (maniac103) [reply mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.java]: (And BTW, you're using incorrect terminology there: the view is already `inflated` (XML description was used to create a view hierarchy), it's just not laid out yet)

File: mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.java [hunk 4]
```diff
@@ -1,13 +1,259 @@
...
+            mapView.setMultiTouchControls(true);
+            mapView.setVerticalMapRepetitionEnabled(false);
+            mapView.getOverlays().add(new CopyrightOverlay(itemView.getContext()));
+            mapView.onResume();
+            applyPositionAndLabelWhenReady(mapView, 16.0f, true, true);
+        }
+
+        private void applyPositionAndLabelWhenReady(MapView mapView, float zoomLevel,
+                boolean allowDrag, boolean allowScroll) {
+            mHandler.post(() -> applyPositionAndLabel(mapView, zoomLevel, allowDrag, allowScroll));
```
Comment (maniac103) [inline mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.java]: Still not too happy with this, but I realize it's an upstream issue. Should probably merge this now and revisit when they fix this.
